pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -56,22 +57,56 @@ type AzVolumeList struct {
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // AzVolumeAttachment is a specification for a AzVolumeAttachment resource
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="NodeName",type=string,JSONPath=`.spec.nodeName`,description="Name of the Node which this AzVolumeAttachment object is attached to"
+// +kubebuilder:printcolumn:name="UnderlyingVolume",type=string,JSONPath=`.spec.underlyingVolume`,description="Name of the Volume which this AzVolumeAttachment object references"
+// +kubebuilder:printcolumn:name="AttachmentTier",type=string,JSONPath=`.spec.attachmentTier`,description="Indicates if the volume attachment should be primary attachment or not"
+// +kubebuilder:printcolumn:name="AttachmentState",type=string,JSONPath=`.status.attachmentState`,description="Indicates if the volume is currently attached or detached to the specified node"
+// +kubebuilder:printcolumn:name="AttachmentTier",type=string,JSONPath=`.status.attachmentTier`,description="Indicates if the volume attachment is primary attachment or not"
 type AzVolumeAttachment struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   AzVolumeAttachmentSpec   `json:"spec"`
-	Status AzVolumeAttachmentStatus `json:"status"`
+	Spec   AzVolumeAttachmentSpec    `json:"spec"`
```

Comment:
It would be good to document the fields as @abhisheksinghbaghel has done for the `AzDriverNode` additions/changes.

Reply:
Done :)